src/App/Fossa/Analyze.hs

```diff
@@ -281,6 +282,14 @@ analyze (BaseDir basedir) destination override unpackArchives jsonOutput enableV
 
 analyzeVSI :: (MonadIO m, Has Diag.Diagnostics sig m, Has Exec sig m, Has (Lift IO) sig m, Has Logger sig m) => VSIAnalysisMode -> Maybe ApiOpts -> Path Abs Dir -> AllFilters -> m (Maybe SourceUnit)
 analyzeVSI VSIAnalysisEnabled (Just apiOpts) dir filters = do
+  -- The endpoint to disable this function if VSI analysis is disabled server-side is new.
+  -- For most feature flags, we'd want to default to the flag *not* enabled.
+  -- However in the case of VSI, we want to default to the flag being enabled if we fail to check it in Core to maintain existing functionality.
+  enabled <- recover $ featureFlagEnabled apiOpts FeatureFlagVSIMonorepo
+  if enabled == Just False
+    then fatalText "VSI analysis is not enabled in FOSSA. Please contact FOSSA for assistance."
```

Comment:
[optional nit/preference] I prefer we use the FF display name here to report error to reduce ambiguity

```
E.g. Feature: Vendored Package Scanning (VPS) is not enabled in FOSSA. Please contact FOSSA. 
```

Ideally we can generate this text dynamically based on FF requested.